### ISSUE TITLE
Handle use of `<>` as brackets

### ISSUE THREAD

**Issue by davidshepherd7** [davidshepherd7/electric-operator#8]
For C/C++ includes, C++ templates, Java generics, ... the `<` and `>` operators are used as brackets.

It might be possible to detect matching pairs and correct the spacing for these cases.

**Comment by davidshepherd7** [comment]
`include` statements are already handled correctly (by checking for a `#include`), but templates/generics are probably impossible to handle without something like this.

**Comment by Dickby** [comment]
In C includes with <> as brackets it's not possible to enter a path, cause a slash gets surrounded by spaces.

**Comment by davidshepherd7** [comment]
Yeah, I think I forgot to handle that. Should be able to fix it tomorrow. Thanks for reporting it :+1

**Comment by davidshepherd7** [comment]
Ok, slashes inside <> includes should be fixed now, let me know if it's not.

**Comment by Dickby** [comment]
It's fixed, thanks a lot.
But there is another problem, I will open a new issue for that.

**Comment by andreas-roehler** [comment]
May an example which triggers the bug be provided?
As for the functionality requested: Melpa delivers packages simple-paren and syntactic-close.

**Comment by andreas-roehler** [comment]
Spotted so far this:

electric-operator-c-pointer-type-style 'variable
  "Defines how C/C++ mode pointer and reference types are spaced.
If set to 'variable' then the operator is touching the variable
name, as in `int *x'.
If set to 'type' then the operator is touching the type name , as
in `int* x'."

;;;;

Is this really a matter of style? I.e. are  `int *x' and `int* x' semantically equal?

**Comment by davidshepherd7** [comment]
> May an example which triggers the bug be provided?

Yeah inside `features/c++-mode.feature` there are a bunch of tests tagged with `known-failure`.

>  As for the functionality requested: Melpa delivers packages simple-paren and syntactic-close.

Thanks, I'll have a look at those when I get a chance.

**Comment by andreas-roehler** [comment]
A generic solution from here might be saying: EO takes action only if no whitespace is before.
